pkg/monitoring/perfscale/vm-status-transitions.go

```diff
@@ -0,0 +1,105 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2020 Red Hat, Inc.
+ *
+ */
+
+package perfscale
+
+import (
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/cache"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	v1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/client-go/log"
+)
+
+func getVMTransitionTimeSeconds(newVM *v1.VirtualMachine) (float64, error) {
+	var oldTime *metav1.Time
+	var newTime *metav1.Time
+
+	// if the vm was deleted, use the deletion timestamp as the latency baseline
+	oldTime = newVM.DeletionTimestamp.DeepCopy()
+	if oldTime == nil {
+		oldTime = newVM.CreationTimestamp.DeepCopy()
+	}
+	for _, transitionTimestamp := range newVM.Status.StatusTransitionTimestamps {
+		if transitionTimestamp.Status == newVM.Status.PrintableStatus {
+			newTime = transitionTimestamp.StatusTransitionTimestamp.DeepCopy()
+		} else if newTime != nil {
+			break
+		}
```

Comment:
Hey @marceloamaral

sorry i'm delayed in giving feedback here. 

I'm unsure if we should be anchoring a metric to thie vm.Status.PrintableStatus value. My concern is that `PrintableStatus` is only meant to serve as a best effort representation of the state a VM is that is used to provide user feedback via CLI and UI.

PrintableStatus is not a programmatic interface that is guaranteed to behave in any sort of consistent way throughout kubevirt releases. It's possible for this field to get new values added, removed, modified, etc... We maintain the freedom to modify how this field is calculated as well in ways that could impact the transition times in the future.

In contrast to PrintableStatus, the vmi.Status.Phase field is meant to be a programmatic interface that behaves consistently. We made the decision to limit the phases to only the values that exist today, and to express any additional information outside of the phases using conditions. This is why having a metric associated with vmi.Status.Phase is likely consistent throughout future kubevirt releases.

The other kind of weird thing about PrintableStatus is that unlike vmi.Status.Phase, we can re-encounter transitioning to the same PrintableStatus many times throughout a VM's lifecycle. Even during the start flow, if an transient error is encountered, it's possible for the PrintableStatus field to transition to the same value multiple times before hitting "Running"

What metric is most important to you here? Perhaps there's another way to track it?